openmetadata-ui/src/main/resources/ui/src/components/common/DeleteWidget/DeleteWidget.interface.ts

```diff
@@ -11,7 +11,12 @@
  *  limitations under the License.
  */
 
-export interface DeleteWidgetModalProps {
+import { AxiosResponse } from 'axios';
+import { Table } from 'generated/entity/data/table';
+
+export type DeleteUnion = Table;
+export interface DeleteWidgetModalProps<T> {
+  api?: () => Promise<AxiosResponse<T>>;
```

Comment:
I don't think we need to do this, we are just creating an extra prop, because anyway, we have to validate the response on the component where we are using the delete widget modal.

Reply:
@Sachin-chaurasiya  we have reverted the  use to DeleteWidgetModal to EntityDeleteModal.